Squared/Threading/Future.cs

```diff
@@ -254,6 +258,10 @@ private sealed class WaitForFirstThunk {
             public WaitForFirstThunk (IEnumerable<IFuture> futures) {
                 OnFutureResolved handler = null;
 
+                // REVIEW: On a theoretical level, I am worried that, if the enumerable is
+                // empty and not detected by the condition below, or if all futures are null
+                // then Composite never completes.
+                // On a practical level, I suspect that it has not happened often. ;)
```

Comment:
Good point, I should make sure that it works correctly without hitting this. I think it does, but it may not have test coverage.